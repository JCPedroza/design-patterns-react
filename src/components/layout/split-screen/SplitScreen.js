import { arrayOf, element, number } from 'prop-types'
import styled from 'styled-components'

const Container = styled.div`
  display: flex;
`

const Pane = styled.div`
  flex: ${props => props.weight};
`

const SplitScreen = ({ children, leftWeight = 1, rightWeight = 1 }) => {
  const [left, right] = children

  return (
    <Container>
      <Pane weight={leftWeight}>
        {left}
      </Pane>

      <Pane weight={rightWeight}>
        {right}
      </Pane>
    </Container>
  )
}

SplitScreen.propTypes = {
  children: arrayOf(element).isRequired,
  leftWeight: number,
  rightWeight: number
}

export default SplitScreen
