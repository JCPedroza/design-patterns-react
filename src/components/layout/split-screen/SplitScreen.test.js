import { render } from '@testing-library/react'
import SplitScreen from './SplitScreen'

const Left = () => <p>left</p>
const Right = () => <p>right</p>
const weights = [3, 5]

describe('SplitScreen component', () => {
  it('renders left and right components', () => {
    const { container } = render(
      <SplitScreen weights={weights}>
        <Left />
        <Right />
      </SplitScreen>
    )
    expect(container).toMatchSnapshot()
  })
})
