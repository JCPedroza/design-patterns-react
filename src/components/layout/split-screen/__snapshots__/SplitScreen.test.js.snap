// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitScreen component renders left and right components 1`] = `
<div>
  <div
    class="sc-bdvvtL jxnajB"
  >
    <div
      class="sc-gsDKAQ eSVvJK"
    >
      <p>
        left
      </p>
    </div>
    <div
      class="sc-gsDKAQ eSVXRk"
    >
      <p>
        right
      </p>
    </div>
  </div>
</div>
`;
